### PR TITLE
feat: OptimizerFocusSearch

### PR DESCRIPTION
**Pull request by sumny**
Reimplementation of https://github.com/berndbischl/focussearch
Automatically performs restarts once `maxit` is exceeded (depending on the `Terminator`)

### CODE REVIEW

**Comment by be-marc** [comment]
Looks good. Can you copy the `FIXME`s to issues, please? Do we have to find a solution for `# FIXME: cannot shrink ParamLgl due to different $levels active binding` before merging?

**Comment by sumny** [comment]
> Looks good. Can you copy the `FIXME`s to issues, please? Do we have to find a solution for `# FIXME: cannot shrink ParamLgl due to different $levels active binding` before merging?

thanks for your review!
Regrading `ParamLgl` and shrinking - I wanted to discuss this anyhow (i.e., if we really want to shrink to only a single "level").
Technically this is currently not that easy in `paradox`, due to `levels` in `ParamFct` being a public field but in `ParamLgl` being an AB set to `levels = function() c(TRUE, FALSE)` (https://github.com/mlr-org/paradox/blob/69c7fd4a876d94a469c2e49292ced17c3af10256/R/ParamLgl.R#L31)

**Comment by jakob-r** [comment]
Könntest du nicht einfach die `ps$vals` setzten für fct und lgl? dann beim samplen einfach entweder a) die samples mit den fixen werten überschreiben oder b) ein param subset bauen aus dem du samplest.

Hast du mal ein benchmark gemacht? Die Sampler sind sehr langsam wenn ich mich richtig erinnere.

... sry forogt to start to think in english.

**Comment by sumny** [comment]
Reiterated.
Now sets a default value for `ParamLgl` (and adds a tag `shrinked`) to indicate that values should be overwritten by this default after sampling.
Also constructs the `ps_local` from scratch to no longer rely on the backend.

File: R/OptimizerFocusSearch.R [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+#' @template example
+OptimizerFocusSearch = R6Class("OptimizerFocusSearch",
+  inherit = Optimizer,
+  public = list(
+
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      # Note: maybe make range / 2 a hyperparameter?
+      ps = ParamSet$new(list(
```
Comment (be-marc) [inline R/OptimizerFocusSearch.R:43]: New paradox sugar

File: R/OptimizerFocusSearch.R [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+#' (currently not chosen) level is dropped.
+#'
+#' Note that for [paradox::ParamLgl]s the value to be shrinked around is set as
+#' the `default` value instead of dropping a level. Also, a tag `shrinked` is
+#' added.
+#'
+#' If the [paradox::ParamSet] has a trafo, `x` is expected to contain the
+#' transformed values.
+#'
+#' @param ps ([paradox::ParamSet])\cr
```
Comment (be-marc) [inline R/OptimizerFocusSearch.R:126]: Use `param_set` instead of `ps` since `ps()` is a new paradox function.

**Pull request merged**